### PR TITLE
Raise a warning when validating extensions that are not e10s compatible

### LINKED ISSUE #433: Raise a warning if not e10s compatible
If an add-on is not e10s compatible as marked by: https://developer.mozilla.org/en-US/Add-ons/Install_Manifests#multiprocessCompatible then raise a warning.

We will likely elevate this to an error at a later date.

Supports: mozilla/addons-server#2770

### PR DESCRIPTION
**Pull request by diox**
Fix #433

### CODE REVIEW

**Comment by kewisch** [comment]
Does this cover products that do not make use of e10s, for example Thunderbird? There should not be a warning for add-ons that support only such products.

**Comment by wagnerand** [comment]
Looks like this is covered in https://github.com/mozilla/amo-validator/pull/434/files#diff-dc1d2989d4782f40310ea008f5f92f36R388. @diox can confirm.

**Comment by diox** [comment]
Yes, I made sure to only warn if the extension targets Firefox. I forgot to add a test for this, though, I'll do that.

File: tests/test_validate.py [hunk 1]
```diff
@@ -105,3 +103,144 @@ def test_is_compat():
...
+    assert data['compatibility_summary']['errors'] == 0
+    assert data['compatibility_summary']['notices'] == 0
+    assert data['compatibility_summary']['warnings'] == 0
+    assert data['detected_type'] == 'extension'
+    assert data['messages'] == []
+    assert data['message_tree'] == {}
+    assert data['signing_summary']['high'] == 0
+    assert data['signing_summary']['medium'] == 0
+    assert data['signing_summary']['low'] == 0
+    assert data['signing_summary']['trivial'] == 0
```
Comment (EnTeQuAk) [inline tests/test_validate.py]: :heart:

**Comment by EnTeQuAk** [comment]
Looks good to me, let me know when you add the additional test and I'll have another look.

**Comment by diox** [comment]
@EnTeQuAk test added

**Comment by EnTeQuAk** [comment]
r+

**Pull request merged**